simplified-app-shared/src/main/java/org/nypl/simplified/app/MainSettingsAccountActivity.java

```diff
@@ -941,6 +981,26 @@ public void run() {
             in_pin_text.setText(creds.getPin().toString());
             in_pin_text.setContentDescription(creds.getPin().toString().replaceAll(".(?=.)", "$0,"));
 
+            if (account.supportsBarcodeDisplay()) {
```

Comment:
I would say to add the conditional that generateBarcodeImage doesn't fail. We don't need any of the objects visible or setting onClick listeners if we know there's not going to be a valid image created.

This will be fairly common trying to use an alphanumeric "barcode" which is probably a good percentage of users in certain library systems where usernames have been in common use for awhile. The variable `barcode` should really be updated to `username` but I'll get to that another time.

Reply:
See a5f2ceb